lib/manager.rb

```diff
@@ -0,0 +1,77 @@
+# Require gems
+require "date"
+
+# Require relatives
+require_relative "room.rb"
+require_relative "reservation.rb"
+
+module Hotel
+  class Manager
+    attr_reader :rooms, :reservations
+
+    def initialize
+      @rooms = (1..20).map do |room_number|
+        Hotel::Room.new(room_number: room_number)
+      end
+
+      @reservations = []
+    end
+
+    def reserve_room(check_in, check_out, room_selection: nil)
+      available_room_list = list_available_rooms(check_in, check_out)
+
+      if available_room_list == []
+        return available_room_list
```

Comment:
I'm not sure that `return`ing an empty array is the right approach here. You might consider either raising an exception or returning `nil`.